applications/serial_lte_modem/src/slm_at_udp_proxy.c

```diff
@@ -191,6 +192,25 @@ static int do_udp_client_connect(const char *url, uint16_t port)
 			ret = -errno;
 			goto cli_exit;
 		}
+
+		/* Best effort to enable/disable DTLS conn_id and load it */
+		uint32_t dtls_cid = (proxy.conn_id != 0) ?
+			NRF_SO_SEC_DTLS_CID_ENABLED : NRF_SO_SEC_DTLS_CID_DISABLED;
+
+		ret = nrf_setsockopt(proxy.sock, NRF_SOL_SECURE, NRF_SO_SEC_DTLS_CID,
+				     &dtls_cid, sizeof(dtls_cid));
```

Comment:
```suggestion
		if (proxy.conn_id) {
				ret = nrf_setsockopt(proxy.sock, NRF_SOL_SECURE, NRF_SO_SEC_DTLS_CID,
						     &proxy.conn_id, sizeof(proxy.conn_id));
		}				  
```
I propose that we only set the socket option if it is other than zero.
Also require that AT parameter is 0, 1 or 2, so it is already same as what our socket API can take.

Reply:
It is a bit awkward to set TLS_DTLS_CID_SUPPORTED, but will be allowed.
Then we'll need to differentiate TLS_DTLS_CID_SUPPORTED and TLS_DTLS_CID_ENABLED.